controller/src/main/java/io/pravega/controller/util/Config.java

```diff
@@ -210,6 +210,9 @@ public final class Config {
     public static final Property<Integer> PROPERTY_LIST_COMPLETED_TXN_MAX_RECORDS = Property.named(
             "listCompletedTxn.max.records", 500);
 
+    public static final Property<Integer> PROPERTY_BUCKET_DISTRIBUTION_INTERVAL_SECONDS = Property.named(
+            "bucket.distribution.interval.in.seconds", 10);
```

Comment:
PROPERTY_BUCKET_DISTRIBUTION_INTERVAL_SECONDS  is set as 10 seconds so redistribution logic will keep on checking for buckets assigned currently to controller every 10 seconds. 
Will this not impact performance ?
How did we arrive to have interval as 10 sec ?

Reply:
No, redistribution of buckets will not happens after every 10 secs. Redistribution of buckets will happens only when there is any changes in controller instances. Once leader detect any change, it will wait for PROPERTY_BUCKET_DISTRIBUTION_INTERVAL_SECONDS before redistributing the buckets.